gdc_client/upload/client.py

```diff
@@ -564,8 +570,11 @@ def upload_parts(self):
 
             for future in as_completed(future_to_part_number):
                 part_number = future_to_part_number[future]
-                log.debug("Part: {} is done".format(part_number))
-                pbar.update()
+                if future.result():
+                    log.debug("Part: {} is done".format(part_number))
+                    pbar.update()
+                else:
+                    log.warning("Part: {} failed".format(part_number))
```

Comment:
Since the key change in this pr is this logging change, could you add a test that captures the logs and verifies that the correct message is logged?  You should be able to use [caplog](https://docs.pytest.org/en/stable/logging.html#caplog-fixture) to test this.